pyro/contrib/gp/likelihoods/binary.py

```diff
@@ -47,4 +48,4 @@ def forward(self, f_loc, f_var, y=None):
             y_dist = dist.Bernoulli(f_res)
         if y is not None:
             y_dist = y_dist.expand_by(y.shape[:-f.dim()]).to_event(y.dim())
-        return pyro.sample("y", y_dist, obs=y)
+        return pyro.sample(_make_name(self._pyro_name, "y"), y_dist, obs=y)
```

Comment:
Interesting usage pattern, let's try to later refactor this into `PyroModule`. Maybe we could add something like
```py
class Binary(Likelihood):
    def __init__(...):
        ...
        self.y = PyroConditionalSample()
    def forward(...):
        ....
        return self.y(prior=y_dist, obs=y)
```
Anyway this is something to think about...

Reply:
The current PyroModule is already pretty nice and clean. I really like it. I think that it is better to get feedback from users before adding this new pattern for it.